lib/Net/Async/Blockchain/Omni.pm

```diff
@@ -92,29 +92,55 @@ async sub transform_transaction {
 
     # the command listtransactions will guarantee that this transactions is from or to one
     # of the node addresses.
-    my $received_transaction;
-    my $detailed_transaction;
+    my ($omni_transaction, $parent_transaction);
+
     try {
-        $received_transaction = await $self->rpc_client->get_transaction($decoded_raw_transaction->{txid});
-        $detailed_transaction   = await $self->rpc_client->get_detailed_transaction($decoded_raw_transaction->{txid});
+        $omni_transaction   = await $self->rpc_client->get_transaction($decoded_raw_transaction->{txid});
+        $parent_transaction = await $self->rpc_client->get_detailed_transaction($decoded_raw_transaction->{txid});
     }
     catch {
         # transaction not found
         return undef;
     }
 
     # transaction not found, just ignore.
-    return undef unless $received_transaction && $received_transaction->{ismine};
+    return undef unless $omni_transaction && $omni_transaction->{ismine};
 
-    my $amount = Math::BigFloat->new($received_transaction->{amount});
-    my $fee    = Math::BigFloat->new($received_transaction->{fee} // 0);
-    my $block  = Math::BigInt->new($received_transaction->{block});
-    my ($from, $to) =
-        await Future->needs_all(map { $self->rpc_client->validate_address($received_transaction->{$_}) } qw(sendingaddress referenceaddress));
+    my $transaction = await _process_transaction($self, $omni_transaction, $parent_transaction);
 
-    my %category;
+    $self->source->emit($transaction) if $transaction;
+
+    return 1;
+}
+
+=head2 _process_transaction
+
+This will process the transaction.
+
+=cut
+
+async sub _process_transaction {
+    my ($self, $omni_transaction, $parent_transaction) = @_;
+
+    my ($senall_amount, $sendall_property_id);
+
+    if ($omni_transaction->{type} eq "Send All") {
+        for my $data ($omni_transaction->{subsends}->@*) {
+            if ($data->{propertyid} == '31') {
```

Comment:
Please document and/or make into a constant this Magic Number

Reply:
Magic Number is stored in magic hat as constant  😸